src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -219,7 +219,7 @@ internal virtual TypeEnumerator GetTypeEnumerator(Type type, string assemblyFile
             typeFullName = type.FullName;
             TypeEnumerator testTypeEnumerator = GetTypeEnumerator(type, assemblyFileName, discoverInternals, testIdGenerationStrategy);
             ICollection<UnitTestElement>? unitTestCases = testTypeEnumerator.Enumerate(out ICollection<string>? warningsFromTypeEnumerator);
-            bool typeIgnored = ReflectHelper.IsAttributeDefined<IgnoreAttribute>(type, false);
```

Comment:
IsAttributeDefined checks attribute entry by fully qualified assembly name. So it is a non-derived attribute (it needs to be exactly the same type).